app/src/main/java/com/islaharper/dawnofandroid/presentation/screens/login/LoginScreen.kt

```diff
@@ -47,47 +55,123 @@ fun LoginScreen(
                 signedInState = signedInState,
                 messageBarState = messageBarState,
                 onButtonClicked = {
-                    loginViewModel.saveSignedInState(signedIn = true)
-                },
+                    loginViewModel.oneTapSignIn()
+                }
             )
-        },
+        }
+    )
+
+    val launcher = rememberLauncherForActivityResult(
+        ActivityResultContracts.StartIntentSenderForResult()
+    ) { result ->
+        try {
```

Comment:
why do we need a try/catch in this block? what exception are we expecting?

Reply:
Originally I was catching statusCodes for Canceled, NetworkError and an else block, but all three set signedInState to false. 
`catch (e: ApiException) {
            when (e.statusCode) {
                CommonStatusCodes.CANCELED -> {
                    loginViewModel.saveSignedInState(signedIn = false)
                }

                CommonStatusCodes.NETWORK_ERROR -> {
                    loginViewModel.saveSignedInState(signedIn = false)
                }

                else -> {
                    loginViewModel.saveSignedInState(signedIn = false)
                }
            }
        }`

Reply:
Wouldn't we only need try/catch if there is some exception that can be thrown? Is there some documentation or some code you can point me to that explains how the final lambda parameter for `rememberLauncherForActivityResult` works with respect to exceptions?

Reply:
This is where the exceptions were handled from the Google Docs:
https://developers.google.com/identity/one-tap/android/get-saved-credentials#disable-one-tap